Lib/test/test_os.py

```diff
@@ -126,11 +126,21 @@ def test_closerange(self):
     @support.cpython_only
     def test_rename(self):
         path = support.TESTFN
+        not_exists = 'temp2/test'
         old = sys.getrefcount(path)
         self.assertRaises(TypeError, os.rename, path, 0)
         new = sys.getrefcount(path)
         self.assertEqual(old, new)
 
+    def test_rename_bad_dir(self):
+        path1 = 'temp/not-exists'
+        path1new = 'temp/test2/test3/test4'
+        try:
+            os.rename(path1, path1new)
+            self.assertFalse(path.exists(path1new))
+        except FileNotFoundError:
```

Comment:
not sure about this line, but you can use the contextlib.suppress function, like that
```python
with contextlib.suppress(FileNotFoundError):
    os.rename(path1, path1new)
    self.assertFalse(path.exists(path1new))
```